Implement scheduling_endpoint attribute

mcktr opened the pull request:
This implements the attribute `scheduling_endpoint` for hosts and services to show which endpoint is running the scheduler for the check.

fixes #4814

dnsmichi commented:
Thx. Put on hold for now, I'll review after 2.9.

There's schema and backend changes which involve a couple of other projects @lippserd.

dnsmichi commented:
Probably `scheduling_source` or similar would better fit the terminology. I'll look further when 2.10 is near.

mcktr commented:
Yepp, `scheduling_source` sounds more suitable. I'll update the PR when I find a bit time (probably next week).

mcktr commented:
I updated the PR to use the better fitting name.

dnsmichi commented:
@lippserd this is relevant for IcingaDB, please decide on a final name here :)

dnsmichi commented:
@Al2Klimov @N-o-X please discuss this with @lippserd whether it is relevant for the 2.12 RC and IcingaDB.

Al2Klimov commented:
AFAIK we've already implemented all needed attributes for IcingaDB.

Al2Klimov commented on lib/icinga/checkresult.ti:
```diff
@@ -78,6 +78,7 @@ class CheckResult
 	};
 
 	[state] String check_source;
+	[state] String scheduling_source;
```
Don't you need to sync this one across the cluster?

Al2Klimov replied:
Forget that question, check_source isn't explicitly synced either.

mcktr commented:
`scheduling_source` indicates which node is responsible for scheduling the checks, to made this information available for Icinga Web 2

# Test

- master
- agent

```
# /etc/icinga2/zones.d/master/test.conf

object Host "host-scheduled-on-master" {
	address = "172.17.0.2"

	check_command = "random"

	check_interval = 30s
	retry_interval = 10s
	max_check_attempts = 3
}

object Service "service-scheduled-on-master" {
	check_command = "random"

	command_endpoint = "deb10i2b1"

	check_interval = 10s
	retry_interval = 5s
	max_check_attempts = 5

	host_name = "host-scheduled-on-master"
}
```

Querying the API: 

```
{
  "attrs": {
    "__name": "host-scheduled-on-master!service-scheduled-on-master",
    [...]
    "check_source": "deb10i2b1",
    [...]
    "scheduling_source": "metis",
    [...]
}
```

lippserd commented:
@N-o-X or @Al2Klimov Please don't forget to create a follow up issue for Icinga DB. And this PR should also add docs since this is also exposed in our API, right?

nilmerg commented:
@mcktr Any chance that you'll add documentation for this until next monday? I know, short notice, but I'd like to have this included in 2.13 and not post-poned just because a doc update is missing. If you don't have time for this, comment or give me a thumbs-down, then we'll do the doc update. (I'll do it if required, but I'd welcome you doing it, as I don't know much about this or Icinga 2 documentation to begin with :sweat_smile: )

mcktr commented:
@nilmerg I added the missing documentation for the `scheduling_source` attribute. I also rebased this PR against the `master` branch and resolved all merge conflicts. 

Best regards
Michael

Al2Klimov reviewed (changes requested):
One last thing:

Please squash the commits and adjust the attribute name in the commit message.
